### PR TITLE
Simplify BucketedInput serialization

### PR DESCRIPTION
**Pull request by clairemcginty**
Removes overrides of readObject/writeObject by simply making all class members serializable/

### CODE REVIEW

File: scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java [hunk 1]
```diff
@@ -416,14 +415,16 @@ public SourceMetadata<V> getSourceMetadata() {
...
     protected Keying keying;
     // lazy, internal checks depend on what kind of iteration is requested
     protected transient SourceMetadata<V> sourceMetadata = null; // lazy
 
+    private transient Map<ResourceId, KV<String, FileOperations<V>>> inputs;
+    private final Map<Integer, KV<String, FileOperations>> fileOperationsEncoding;
+    private final Map<ResourceId, Integer> directoriesEncoding;
+
     // Used to efficiently serialize BucketedInput instances
     private static Coder<Map<ResourceId, Integer>> directoriesEncodingCoder =
```
Comment (RustedBones) [inline scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java:429]: It looks like `directoriesEncodingCoder` and `fileOperationsEncodingCoder` are not relevant anymore

File: scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java [hunk 2]
```diff
@@ -416,14 +415,16 @@ public SourceMetadata<V> getSourceMetadata() {
...
-
     protected TupleTag<V> tupleTag;
-    protected Map<ResourceId, KV<String, FileOperations<V>>> inputs;
     protected Predicate<V> predicate;
     protected Keying keying;
     // lazy, internal checks depend on what kind of iteration is requested
     protected transient SourceMetadata<V> sourceMetadata = null; // lazy
 
+    private transient Map<ResourceId, KV<String, FileOperations<V>>> inputs;
+    private final Map<Integer, KV<String, FileOperations>> fileOperationsEncoding;
```
Comment (RustedBones) [inline scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java:425]: We can probably give the type param here
```suggestion
    private final Map<Integer, KV<String, FileOperations<V>>> fileOperationsEncoding;
```

**Review by RustedBones — COMMENTED** [review]
There is still something I don't get: How replacing the 
```java
protected Map<ResourceId, KV<String, FileOperations<V>>> inputs;
```
With
```java
private transient Map<ResourceId, KV<String, FileOperations<V>>> inputs;
private final Map<Integer, KV<String, FileOperations>> fileOperationsEncoding;
private final Map<ResourceId, Integer> directoriesEncoding;
```
now makes the class serializable ?

**Comment by clairemcginty** [comment]
> There is still something I don't get: How replacing the
> 
> ```java
> protected Map<ResourceId, KV<String, FileOperations<V>>> inputs;
> ```
> 
> With
> 
> ```java
> private transient Map<ResourceId, KV<String, FileOperations<V>>> inputs;
> private final Map<Integer, KV<String, FileOperations>> fileOperationsEncoding;
> private final Map<ResourceId, Integer> directoriesEncoding;
> ```
> 
> now makes the class serializable ?

Sorry for the confusion. To clarify, BucketedInput was `Serializable`, checks with `SerializableUtils` passed, and Dataflow had no issue parsing it. However, on FlinkRunner, we were seeing this error with Flink's custom serializer:

```
Caused by: java.lang.RuntimeException: Deserializing the input/output formats failed: unread block data
	at org.apache.flink.runtime.jobgraph.InputOutputFormatContainer.<init>(InputOutputFormatContainer.java:69)
	at org.apache.flink.runtime.jobgraph.InputOutputFormatVertex.initInputOutputformatContainer(InputOutputFormatVertex.java:185)
	... 17 more
Caused by: java.lang.IllegalStateException: unread block data
	at java.base/java.io.ObjectInputStream$BlockDataInputStream.setBlockDataMode(ObjectInputStream.java:3033)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1722)
	at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2496)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2390)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2228)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1687)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:489)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:447)
	at org.apache.flink.util.InstantiationUtil.deserializeObject(InstantiationUtil.java:539)
	at org.apache.flink.util.InstantiationUtil.deserializeObject(InstantiationUtil.java:527)
...
```

My suspicion is that we weren't properly flushing something in our complex `readObject`/`writeObject` overrides.  Sure enough, after removing them and delegating to the default Java object serialization, SMB reads now work on FlinkRunner.

It's a bit frustrating because I haven't been able to repro this with a unit test (even using the Flink serializer class...), so it's all been manually tested via deploys to a locally running Flink cluster.

**Review by RustedBones — APPROVED** [review]

**Pull request merged**